### PR TITLE
Feat: Add migrations and seed data

### PR DESCRIPTION
**Pull request by brycedev**
Fixes https://github.com/midday-ai/v1/issues/1

- Create primary migrations (auth.users, public.users, public.posts)
- Add seed data (...)

Figured I would leave display to you @pontusab

### CODE REVIEW

File: apps/api/supabase/migrations/20240901155538_create_users_table.sql [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+end;
+
+$$ language plpgsql;
+
+create trigger users_updated_at before
+update
+  on public .users for each row execute function update_updated_at();
+
+-- create a policy to allow users to read all profiles
+create policy read_all_profiles on public .users for
```
Comment (xyzeva) [inline apps/api/supabase/migrations/20240901155538_create_users_table.sql:39]: // security: allows all users emails to be read. very bad.

File: apps/api/supabase/migrations/20240901155538_create_users_table.sql [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+select
+  using (true);
+
+-- create a policy to allow users to update their own profile
+create policy update_own_profile on public .users for
```
Comment (xyzeva) [inline apps/api/supabase/migrations/20240901155538_create_users_table.sql:44]: // security: there may be sensitive data that we still do not want the user to change in their own user object, possibly add an exclude field list

**Review by xyzeva — CHANGES_REQUESTED** [review]

**Comment by pontusab** [comment]
thank you! I can take over the RLS if you want regarding security!

File: apps/api/supabase/migrations/20240901155538_create_users_table.sql (hunk 1 above)
Comment (brycedev) [reply apps/api/supabase/migrations/20240901155538_create_users_table.sql:39]: @xyzeva I agree. I think the email doesn't need to be on this table at all actually, but was following the user update schema.

File: apps/api/supabase/migrations/20240901155538_create_users_table.sql (hunk 2 above)
Comment (brycedev) [reply apps/api/supabase/migrations/20240901155538_create_users_table.sql:44]: I would think this table doesn't include anything sensitive. This is akin to a "profiles" table; only include data you would want another user to see. Sensitive data should be locked down to auth.users

**Comment by brycedev** [comment]
> thank you! I can take over the RLS if you want regarding security!

All yours!

**Comment by pontusab** [comment]
Closing in favor of: https://github.com/midday-ai/v1/pull/26

**Pull request closed without merging**